src/main/java/com/commercetools/sync/commons/BaseSyncOptions.java

```diff
@@ -96,23 +104,25 @@ public void applyWarningCallback(@Nonnull final String warningMessage) {
      * which is set to {@code this} instance of the {@link BaseSyncOptions}. If there {@code errorCallback} is null,
      * this method does nothing.
      *
-     * @param errorMessage the error message to supply as first param to the {@code errorCallback} function.
-     * @param exception    optional {@link Throwable} instance to supply to the {@code errorCallback} function as a
-     *                     second param.
+     * @param exception {@link Throwable} instance to supply as first param to the {@code errorCallback} function.
+     * @param oldResource the old resource that is being compared to the new draft.
+     * @param newResourceDraft the new resource draft that is being compared to the old resource.
+     * @param updateActions the list of update actions.
      */
-    public void applyErrorCallback(@Nonnull final String errorMessage, @Nullable final Throwable exception) {
+    public void applyErrorCallback(@Nonnull final SyncException exception, @Nullable final U oldResource,
+        @Nullable final V newResourceDraft, @Nullable final List<UpdateAction<U>> updateActions) {
         if (this.errorCallBack != null) {
-            this.errorCallBack.accept(errorMessage, exception);
+            this.errorCallBack.accept(exception, oldResource, newResourceDraft, Optional.ofNullable(updateActions));
         }
     }
 
     /**
      *
-     * @param errorMessage the error message to supply as first param to the {@code errorCallback} function.
-     * @see #applyErrorCallback(String, Throwable) applyErrorCallback(String, Throwable)
+     * @param errorMessage the error message to supply as part of first param to the {@code errorCallback} function.
+     * @see #applyErrorCallback(SyncException, Object, Object, List)
      */
     public void applyErrorCallback(@Nonnull final String errorMessage) {
-        applyErrorCallback(errorMessage, null);
+        applyErrorCallback(new SyncException(errorMessage), null, null, null);
```

Comment:
Currently:
- validating batch of drafts (`BatchProcessor#validateBatch`)
- preparing updateActions (`GenericUpdateActionUtils#buildTypedSetCustomTypeUpdateAction`,  `CustomUpdateActionUtils#buildCustomUpdateActions` and `ProductVariantUpdateActionUtils#buildProductVariantPricesUpdateActions`)

but:
- with providing old/new resource for https://github.com/commercetools/commercetools-sync-java/pull/367#discussion_r249037130 case `ProductVariantUpdateActionUtils#buildProductVariantPricesUpdateActions` will no longer be valid - will change this. 
- usage in `CustomUpdateActionUtils#buildCustomUpdateActions` can also be modified to same as for https://github.com/commercetools/commercetools-sync-java/pull/367#discussion_r249038697 - if condition is true provide old/new resource otherwise null

Reply:
I think definitely it does, but I'm not sure it should be part of this PR